.github/workflows/ghrc-publish.yaml

```diff
@@ -0,0 +1,44 @@
+name: Create and publish a Docker image in ghcr.io
+on:
+  push:
+    branches:
+      - master
+      - main
+    tags:
+      - v*
+env:
+  REGISTRY: ghcr.io
+  REPOSITORY: ${{ github.repository }}
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v12
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Build minipool
+        run: |
+          nix build -L .#minipool
+      - name: Build fedimint-observer
+        run: |
+          nix build -L .#minipool_image
+          echo "minipool_image_tag=$(docker load < result | awk '{ print $3 }')" >> $GITHUB_ENV
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3.2.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I'll probably have to add that secret or do they have a agreement with GH to use the same auth?

Reply:
It works because `GITHUB_TOKEN` has default permissions for GHCR in the same repo/account. If we were pushing to a different registry (Docker Hub, AWS, etc.), we'd need to add a separate secret.